File: move-zeros/move-zeroes.js
'use strict';

const moveZeroes = (nums) => {
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      for (let j = 0; j < i; j++) {
        if (nums[j] === 0) {
          const temp = nums[i];
          nums[i] = nums[j];
          nums[j] = temp;
        }
      }
    }
  }
};

export default moveZeroes;

File: move-zeros/move-zeroes.rb
def move_zeroes(nums)
  for i in 0..nums.length - 1
      if (nums[i] != 0) 
          for j in 0..i
              if (nums[j] == 0)
                  temp = nums[i]
                  nums[i] = nums[j]
                  nums[j] = temp
              end
          end
      end
  end
end